RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -0,0 +1,730 @@
+    if( key != eles_by_oop.end() ) {
+      ptr = key->second;
+    } else {
+      throw cms::Exception("BadElectronKey")
+        << "Original object pointer with key = " << ele.originalObjectRef().key() 
+        << " not found in cache!";
+    }
+  }
+
+  std::array<float, 33> eval;
+  
+  reco::SuperClusterRef sc = ele.superCluster();
+  edm::Ptr<reco::CaloCluster> theseed = sc->seed();
+
+  // SET INPUTS
+  eval[0]  = nVtx_;  
+  eval[1]  = sc->rawEnergy();
+  eval[2]  = sc->position().Eta();
+  eval[3]  = sc->position().Phi();
+  eval[4]  = sc->etaWidth();
+  eval[5]  = sc->phiWidth(); 
+  eval[6]  = ele.full5x5_r9();
+  eval[7]  = theseed->energy()/sc->rawEnergy();
+  
+  float sieip, cryPhi, cryEta; 
+  int iPhi=0, iEta=0;
+  float eMax, e2nd, eTop, eBottom, eLeft, eRight;
+  float clusterMaxDR, clusterMaxDRDPhi, clusterMaxDRDEta, clusterMaxDRRawEnergy;
+  float clusterRawEnergy0, clusterRawEnergy1, clusterRawEnergy2;
+  float clusterDPhiToSeed0, clusterDPhiToSeed1, clusterDPhiToSeed2;
+  float clusterDEtaToSeed0, clusterDEtaToSeed1, clusterDEtaToSeed2;
+
+  assignValue(ptr, e_conf.tok_sigmaIetaIphi, ele_vmaps, sieip);
+  assignValue(ptr, e_conf.tok_eMax, ele_vmaps, eMax);
+  assignValue(ptr, e_conf.tok_e2nd, ele_vmaps, e2nd);
+  assignValue(ptr, e_conf.tok_eTop, ele_vmaps, eTop);
+  assignValue(ptr, e_conf.tok_eBottom, ele_vmaps, eBottom);
+  assignValue(ptr, e_conf.tok_eLeft, ele_vmaps, eLeft);
+  assignValue(ptr, e_conf.tok_eRight, ele_vmaps, eRight);
+  assignValue(ptr, e_conf.tok_clusterMaxDR	    , ele_vmaps, clusterMaxDR	    );
+  assignValue(ptr, e_conf.tok_clusterMaxDRDPhi	    , ele_vmaps, clusterMaxDRDPhi	    );
+  assignValue(ptr, e_conf.tok_clusterMaxDRDEta	    , ele_vmaps, clusterMaxDRDEta	    );
+  assignValue(ptr, e_conf.tok_clusterMaxDRRawEnergy , ele_vmaps, clusterMaxDRRawEnergy); 
+  assignValue(ptr, e_conf.tok_clusterRawEnergy0	    , ele_vmaps, clusterRawEnergy0    );	    
+  assignValue(ptr, e_conf.tok_clusterRawEnergy1	    , ele_vmaps, clusterRawEnergy1    );	    
+  assignValue(ptr, e_conf.tok_clusterRawEnergy2	    , ele_vmaps, clusterRawEnergy2    );	    
+  assignValue(ptr, e_conf.tok_clusterDPhiToSeed0    , ele_vmaps, clusterDPhiToSeed0   ); 
+  assignValue(ptr, e_conf.tok_clusterDPhiToSeed1    , ele_vmaps, clusterDPhiToSeed1   ); 
+  assignValue(ptr, e_conf.tok_clusterDPhiToSeed2    , ele_vmaps, clusterDPhiToSeed2   ); 
+  assignValue(ptr, e_conf.tok_clusterDEtaToSeed0    , ele_vmaps, clusterDEtaToSeed0   ); 
+  assignValue(ptr, e_conf.tok_clusterDEtaToSeed1    , ele_vmaps, clusterDEtaToSeed1   ); 
+  assignValue(ptr, e_conf.tok_clusterDEtaToSeed2    , ele_vmaps, clusterDEtaToSeed2   ); 
+  assignValue(ptr, e_conf.tok_iPhi, ele_int_vmaps, iPhi);
+  assignValue(ptr, e_conf.tok_iEta, ele_int_vmaps, iEta);
+  assignValue(ptr, e_conf.tok_cryPhi, ele_vmaps, cryPhi);
+  assignValue(ptr, e_conf.tok_cryEta, ele_vmaps, cryEta);
+
+  eval[8]  = eMax/sc->rawEnergy();
```

Comment:
@slava77 added to nearer-term stack. I'll see what I can do.

On Mon, Aug 10, 2015 at 1:51 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc
> <https://github.com/cms-sw/cmssw/pull/10613#discussion_r36623288>:
>
> > +  assignValue(ptr, e_conf.tok_clusterMaxDRRawEnergy , ele_vmaps, clusterMaxDRRawEnergy);
> > +  assignValue(ptr, e_conf.tok_clusterRawEnergy0	    , ele_vmaps, clusterRawEnergy0    );	
> > +  assignValue(ptr, e_conf.tok_clusterRawEnergy1	    , ele_vmaps, clusterRawEnergy1    );	
> > +  assignValue(ptr, e_conf.tok_clusterRawEnergy2	    , ele_vmaps, clusterRawEnergy2    );	
> > +  assignValue(ptr, e_conf.tok_clusterDPhiToSeed0    , ele_vmaps, clusterDPhiToSeed0   );
> > +  assignValue(ptr, e_conf.tok_clusterDPhiToSeed1    , ele_vmaps, clusterDPhiToSeed1   );
> > +  assignValue(ptr, e_conf.tok_clusterDPhiToSeed2    , ele_vmaps, clusterDPhiToSeed2   );
> > +  assignValue(ptr, e_conf.tok_clusterDEtaToSeed0    , ele_vmaps, clusterDEtaToSeed0   );
> > +  assignValue(ptr, e_conf.tok_clusterDEtaToSeed1    , ele_vmaps, clusterDEtaToSeed1   );
> > +  assignValue(ptr, e_conf.tok_clusterDEtaToSeed2    , ele_vmaps, clusterDEtaToSeed2   );
> > +  assignValue(ptr, e_conf.tok_iPhi, ele_int_vmaps, iPhi);
> > +  assignValue(ptr, e_conf.tok_iEta, ele_int_vmaps, iEta);
> > +  assignValue(ptr, e_conf.tok_cryPhi, ele_vmaps, cryPhi);
> > +  assignValue(ptr, e_conf.tok_cryEta, ele_vmaps, cryEta);
> > +
> > +  eval[8]  = eMax/sc->rawEnergy();
>
> enums maybe?
> We were discussing with Lindsey in another PR to have something like
> TMVAEvaluator for this kind of data manipulation.
> I guess a solution there is coming soon.
> @lgray <https://github.com/lgray>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/10613/files#r36623288>.
>